broker/test/test_broker/eventmesh.LockManager.spec.js

```diff
@@ -328,16 +328,17 @@ describe('eventmesh', () => {
       it('should fail to unlock resource after multiple retries', () => {
         const payload1 = {
           metadata: {
-            resourceVersion: samplelock6.metadata.resourceVersion
+            resourceVersion: samplelock4.metadata.resourceVersion
           }
         };
-        mocks.apiServerEventMesh.nockPatchResource(CONST.APISERVER.RESOURCE_GROUPS.LOCK, CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT_LOCKS, 'samplelock6', samplelock6, 3, payload1, 500);
-        return lockManager.unlock('samplelock6', samplelock6.metadata.resourceVersion, 3, 100)
+        mocks.apiServerEventMesh.nockPatchResource(CONST.APISERVER.RESOURCE_GROUPS.LOCK, CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT_LOCKS, 'samplelock4', samplelock4, 3, payload1, 500);
+        return lockManager.unlock('samplelock4', samplelock4.metadata.resourceVersion, 3, 100)
+          .then(() => assert(false))
           .catch(err => {
```

Comment:
We want our promise to be rejected but what if due to some bug it gets resolved(like it was happening after the nock upgrade), the UT still gets passed when it should fail. To overcome that, we need to explicitly fail the UT in case the promise gets resolved